go/arrow/flight/flightsql/client.go

```diff
@@ -491,6 +490,33 @@ func (p *PreparedStatement) ExecuteUpdate(ctx context.Context) (nrecords int64,
 	return updateResult.GetRecordCount(), nil
 }
 
+func (p *PreparedStatement) hasBindParameters() bool {
+	return (p.paramBinding != nil && p.paramBinding.NumRows() > 0) || (p.streamBinding != nil)
+}
+
+func (p *PreparedStatement) writeBindParameters(pstream pb.FlightService_DoPutClient, desc *pb.FlightDescriptor) (*flight.Writer, error) {
+	if p.paramBinding != nil {
+		wr := flight.NewRecordWriter(pstream, ipc.WithSchema(p.paramBinding.Schema()))
+		wr.SetFlightDescriptor(desc)
+		if err := wr.Write(p.paramBinding); err != nil {
+			return nil, err
+		}
+		return wr, nil
+	} else {
+		wr := flight.NewRecordWriter(pstream, ipc.WithSchema(p.streamBinding.Schema()))
+		wr.SetFlightDescriptor(desc)
+		for p.streamBinding.Next() {
+			if err := wr.Write(p.streamBinding.Record()); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
`arrio.Copy` ?

Reply:
Unfortunately that takes a Reader, but streamBinding is a RecordReader